source/common/network/dns_impl.cc

```diff
@@ -224,5 +212,82 @@ void DnsResolverImpl::PendingResolution::getHostByName(int family) {
       this);
 }
 
+ActiveDnsQuery* DnsResolverImpl::resolveSrv(const std::string& dns_name,
+                                            DnsLookupFamily dns_lookup_family,
+                                            ResolveSrvCb callback) {
+  std::unique_ptr<PendingSrvResolution> pending_srv_res(
+      new PendingSrvResolution(callback, dispatcher_, channel_, dns_name, dns_lookup_family, this));
+  pending_srv_res->getSrvByName();
+  return markForDestruction(std::move(pending_srv_res));
+}
+
+void DnsResolverImpl::PendingSrvResolution::onAresSrvStartCallback(int status, int timeouts,
+                                                                   unsigned char* buf, int len) {
+  // We receive ARES_EDESTRUCTION when destructing with pending queries.
+  if (status == ARES_EDESTRUCTION) {
+    ASSERT(owned_);
+    delete this;
+    return;
+  }
+
+  bool replies_parsed = false;
+  if (status == ARES_SUCCESS) {
+    struct ares_srv_reply* srv_reply;
+    status = ares_parse_srv_reply(buf, len, &srv_reply);
+
+    if (status == ARES_SUCCESS) {
+      std::list<Address::SrvInstanceConstSharedPtr> srv_records;
+      size_t total = 0, finished = 0;
+      for (ares_srv_reply* current_reply = srv_reply; current_reply != NULL;
+           current_reply = current_reply->next, ++total) {
+        resolver_->resolve(
+            current_reply->host, this->dns_lookup_family_,
+            [=, &finished,
+             &srv_records](const std::list<Address::InstanceConstSharedPtr>&& address_list) {
+              for (auto instance = address_list.begin(); instance != address_list.end();
+                   ++instance) {
+                Address::InstanceConstSharedPtr inst_with_port(
+                    Utility::getAddressWithPort(*instance->get(), current_reply->port));
+                srv_records.emplace_back(new Address::SrvInstanceImpl(inst_with_port));
+              }
+              if (++finished == total) {
+                this->onAresSrvFinishCallback(std::move(srv_records));
```

Comment:
`this->` is probably redundant here and elsewhere.

Reply:
Addressed in dc152d2.